osu.Game/Rulesets/Objects/BezierConverter.cs

```diff
@@ -0,0 +1,276 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Utils;
+using osu.Game.Rulesets.Objects.Types;
+using osuTK;
+
+namespace osu.Game.Rulesets.Objects
+{
+    public static class BezierConverter
+    {
+        private struct CircleBezierPreset
+        {
+            public readonly double ArcLength;
+            public readonly Vector2d[] ControlPoints;
+
+            public CircleBezierPreset(double arcLength, Vector2d[] controlPoints)
+            {
+                ArcLength = arcLength;
+                ControlPoints = controlPoints;
+            }
+        }
+
+        // Extremely accurate a bezier anchor positions for approximating circles of several arc lengths
+        private static readonly CircleBezierPreset[] circle_presets =
+        {
+            new CircleBezierPreset(0.4993379862754501,
+                new[] { new Vector2d(1, 0), new Vector2d(1, 0.2549893626632736f), new Vector2d(0.8778997558480327f, 0.47884446188920726f) }),
+            new CircleBezierPreset(1.7579419829169447,
+                new[] { new Vector2d(1, 0), new Vector2d(1, 0.6263026f), new Vector2d(0.42931178f, 1.0990661f), new Vector2d(-0.18605515f, 0.9825393f) }),
+            new CircleBezierPreset(3.1385246920140215,
+                new[] { new Vector2d(1, 0), new Vector2d(1, 0.87084764f), new Vector2d(0.002304826f, 1.5033062f), new Vector2d(-0.9973236f, 0.8739115f), new Vector2d(-0.9999953f, 0.0030679568f) }),
+            new CircleBezierPreset(5.69720464620727,
+                new[] { new Vector2d(1, 0), new Vector2d(1, 1.4137783f), new Vector2d(-1.4305235f, 2.0779421f), new Vector2d(-2.3410065f, -0.94017583f), new Vector2d(0.05132711f, -1.7309346f), new Vector2d(0.8331702f, -0.5530167f) }),
+            new CircleBezierPreset(2 * Math.PI,
+                new[] { new Vector2d(1, 0), new Vector2d(1, 1.2447058f), new Vector2d(-0.8526471f, 2.118367f), new Vector2d(-2.6211002f, 7.854936e-06f), new Vector2d(-0.8526448f, -2.118357f), new Vector2d(1, -1.2447058f), new Vector2d(1, 0) })
+        };
+
+        public static IEnumerable<Vector2> ConvertToLegacyBezier(IList<PathControlPoint> controlPoints, Vector2 position)
```

Comment:
This doesn't have test coverage, but at very least xmldoc would be appreciated on these `public` methods to describe what they are meant to achieve.